core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -90,6 +90,26 @@ import scala.collection.{Map, Seq, Set, immutable, mutable}
 import scala.util.{Failure, Success, Try}
 import kafka.coordinator.group.GroupOverview
 
+case class OnLeadershipChange(groupCoordinator: GroupCoordinator, txnCoordinator: TransactionCoordinator) {
```

Comment:
It feels awkward to have this at the start of KafkaApis.  It also probably shouldn't be a class-- we just care about the function in the class.  So I'd propose just putting this in ApisUtils.

I have a second comment which is that eventually we're going to have to get rid of these "iterate over every partition" kind of functions in the broker.  But now is not the time to do this-- I realize you're working with the existing replica manager APIs.  :)